xbmc/filesystem/SpecialProtocol.cpp

```diff
@@ -28,12 +28,402 @@
 #include "settings/Settings.h"
 #include "utils/log.h"
 #include "utils/URIUtils.h"
-
+#include "Application.h"
+#include "filesystem/Directory.h"
 #include <cassert>
+
 #ifdef TARGET_POSIX
 #include <dirent.h>
 #include "utils/StringUtils.h"
+#include "CompileInfo.h"
+#endif
+
+#ifdef TARGET_WINDOWS
+#include "platform/win32/WIN32Util.h"
+#endif
+
+#ifdef TARGET_DARWIN
+#include "platform/darwin/DarwinUtils.h"
+#endif
+
+#ifdef TARGET_ANDROID
+#include "platform/android/activity/XBMCApp.h"
+#include <androidjni/ApplicationInfo.h>
+#include "platform/android/bionic_supplement/bionic_supplement.h"
+#endif
+
+using namespace XFILE;
+
```

Comment:
You can stick internal helper functions here like this
```
namespace
{
std::string GetLowerAppName()
{
  std::string appName = CCompileInfo::GetAppName();
  std::string dotLowerAppName = "." + appName;
  StringUtils::ToLower(dotLowerAppName);
  return dotLowerAppName;
}
}
```

instead of having them inside the class. This will hide them inside this file only and is a nice way to hide helper functions.